Various fixes before TR11

yordanalexandrov opened the pull request:
+ removed emitting on every form change in column select
+ removed PageData from paginator models
+ replaced PageData with PaginatorData

yordanalexandrov committed:
Various fixes before TR11

+ removed emitting on every form change in column select
+ removed PageData from paginator models
+ replaced PageData with PaginatorData

svilenvelikov commented on cypress/steps/search-column-selector-steps.ts:
```diff
@@ -20,6 +20,10 @@ export class SearchColumnSelectorSteps {
   }
 
+  static closeColumnSelectDropdown() {
+    cy.get('.cdk-overlay-backdrop').click(-50, -50, {force: true});
```
:D It's OK this way. I just wonder if you need the coordinates and the force in the click here?!

yordanalexandrov replied:
Not sure if needed. Comes straight from SO.

svilenvelikov commented on cypress/test/search-column-selector.spec.ts:
```diff
@@ -49,6 +47,7 @@ describe('Onto Search Column Selector Component', () => {
     step.getSelectedOptions().should('have.length', 1).and('contain', 'Name');
+    step.closeColumnSelectDropdown();
     // And the selected columns should be the one passed to component ('Name')
     step.getSelectedColumns().should('contain', 'Name');
```
I ususally try to close menus at the end of the tests. Not a big deal but it might cause troubles in some rare occasions

yordanalexandrov replied:
I close it here because the selection is emitted on panel closed.

yordanalexandrov committed:
Various fixes before TR11

+ removed emitting on every form change in column select
+ removed PageData from paginator models
+ replaced PageData with PaginatorData

yordanalexandrov committed:
Various fixes before TR11

+ removed ngDoCheck from date facet and added ngOnChange check
